Add in string extract logic

bkearney opened the pull request:
This is 4 commits that do the following:
1) Extract the strings and translations for the main code base.
2) Add automation to do (1)
3) Extract the strings and translations for the bastion bits
4) Adds automation for all of the above.
I dont like  the bastion automation since it does not use edit files like the other bits, but I think it is ok. I have copied the items for (1) from the work @lzap did in foreman_discovery. Once this is commited, I will set up transifex to pull from this repo and then ask that every release of katello extract strings and translations as part of the release process. I will set up a seperate process to push downstream strings into trransifex.

akofink commented:
> ask that every release of katello extract strings and translations as part of the release process

Submit a PR [here](https://github.com/theforeman/theforeman.org/blob/gh-pages/plugins/katello/developers.md#how-to-release-katello) if you'd like to formally document this.

bkearney commented:
@akofink will do once reviewed
Also, I will squash this one reviewed.

akofink commented on locale/Makefile:
```diff
@@ -1,41 +1,76 @@
+
+mo-files: $(MOFILES)
+	git add $(POFILES) $(POTFILE) ../locale/*/LC_MESSAGES
+	git commit -m "i18n - pulling from tx"
```
I think it'd be better to make this optional or interactive, at least (y/n). Here's why: most of us are running this from a vagrant box; rake must be run on the vagrant box with the environment (gems, etc.) set up, but git commits must be run from the hypervisor in order to use the proper git user and config (gpg signing, etc.).

bkearney replied:
This is  a copy/paste for the other makefiles. I would prefer to keep them consistent, but I dont do builds. lets ping @ares @lzap @ehelms @dLobatog for their takes on consistency vs the pain of the commit (which I hate btw)

ehelms replied:
I think this is fine given when this is typically run. We can't be sure where a user is running their git code, whether they have it synced or are using it directly from Vagrant (which I do). So let's keep it as is and let releasers figure it out over time.

ehelms commented on update-i18n:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Should this live in locale/ with the Makefile?

bkearney replied:
this was where @lzap put it in his plugins. I am happy to move it, but would prefer consistency

lzap replied:
Yeah locale/ perhaps could have been better fit, but I also do have ./release-gem big script in root that does the whole job of releasing the gem, that one fits better in root, maybe ./extra. Dunno. Honestly, if you do not plan to use this script, I suggest to exclude it. I rarely use it myself.

bkearney replied:
@ehelms  want me to nuke it?

ehelms replied:
The script sounds useful, I still think it'd be cleaner in locale/ and updated for that path to not clutter top level workspace.

ehelms commented:
I'm a little surprised this Makefile is not some set of common functionality provided by Foreman given every plugin runs the same scripts.

ehelms commented:
Is it normal to commit the binary `.mo` files? I thought those were generated during packaging build time.

ehelms commented:
[test katello]

bkearney commented:
@ehelms I dont have a clear answer on the makefile, That is the pattern in all the plugins.

akofink commented:
@bkearney ehelms said the Makefile was fine, so I'm fine with it.

bkearney commented:
@ehelms  I also removed the .mo files and added the ignores to the .gitignore file. This puts this in line with the foreman project.

bkearney commented:
Squashed them all to get around the hound error.

bkearney commented:
I rebased this off of master, and removed the actual new strings since they were making it hard to merge back into master. Once this is merged, I would suggest running this once to refresh the strings and the translations.

beav commented:
[test katello]

beav commented:
@lzap @ehelms @akofink this PR passes automation and I don't think its waiting on anything at this point. Does it need anything else before ack?

lzap commented:
No comments.

beav commented:
@akofink @lzap thanks!

setting waiting on contributor for resolution of https://github.com/Katello/katello/pull/7098#discussion_r192776241

bkearney commented:
@beav moved the script per @ehelms comment.

ehelms commented on update-i18n:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+FOREMAN_DIR="../foreman"
+BASTION_DIR="katello/engines/bastion_katello"
```
Ignore, you just need to remove this file all together now that you moved it.

ehelms commented on locale/update-i18n:
```diff
@@ -0,0 +1,22 @@
+# Now extract the bastion files
+pushd $BASTION_DIR
+bundle exec grunt i18n:extract
+bundle exec grunt i18n:compile
```
Does this work? grunt is a JS based command line so bundle exec doesn't provide any value. I guess the bundle exec does not hurt but can be dropped for clarity.

akofink replied:
I agree, let's remove `bundle exec` here.

ehelms commented on locale/update-i18n:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Just checking that this script assumes you've `bundle installed` from `foreman` and done an `npm install` inside `katello/engines/bastion_katello` ?

thomasmckay commented:
Bump on this, ready to merge?

beav commented:
@ehelms it looks like you have two open questions still, are these blocking from merging or just general q's?

johnsonm325 commented:
@bkearney Just following up on the outstanding questions here. Removing ```bundle exec``` from ```locale/update-i18n``` and checking on a response to @ehelms question, "Just checking that this script assumes you've ```bundle installed``` from ```foreman``` and done an ```npm install``` inside ```katello/engines/bastion_katello```?"

beav commented:
This has been approved for awhile but was waiting on a few additional comments that were not marked as blocking. I'm going to merge now.
